strawberryfields/plot.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This module provides tools to visualize the state in various interactive
+ways using Plot.ly
+"""
+from copy import deepcopy
+
+import numpy as np
+import plotly.io as pio
+
+
+def plot_wigner(state, renderer="browser", wire=0, dq=0.1):
+    """Plot the Wigner function with Plot.ly
+
+    Args:
+        state (:class:`.BaseState`): the state used for plotting
+        renderer (string): the renderer for plotting with Plot.ly
+        wire (int): wire used to calculate the reduced Wigner function
+        dq (float): stepsize along the q and p axes
+    """
+    q = np.arange(-state.cutoff_dim, state.cutoff_dim, dq)
+    pio.renderers.default = renderer
+
+    data = state.wigner(wire, q, q)
+    new_chart = generate_wigner_chart(data, q)
+    pio.show(new_chart)
+
+
+def generate_wigner_chart(data, qp):
+    """Populates a chart dictionary with reduced Wigner function surface plot data.
+
+    Args:
+        qp (tuple[float]): minimum and maximum values of the q and p axes
```

Comment:
I wonder how easy it would be to have this accept q and p axis list values as well 🤔